test/testutil/test_utils.go

```diff
@@ -0,0 +1,1107 @@
+package testutil
+
+import (
+	"errors"
+	"math/rand"
+	"reflect"
+	"slices"
+	"strconv"
+	"time"
+
+	"cosmossdk.io/core/header"
+	"cosmossdk.io/core/store"
+	"cosmossdk.io/log"
+	cosmosMath "cosmossdk.io/math"
+	storetypes "cosmossdk.io/store/types"
+	"github.com/cometbft/cometbft/crypto/secp256k1"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codecAddress "github.com/cosmos/cosmos-sdk/codec/address"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	"github.com/cosmos/cosmos-sdk/testutil"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	authcodec "github.com/cosmos/cosmos-sdk/x/auth/codec"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/allora-network/allora-chain/app/params"
+	alloralog "github.com/allora-network/allora-chain/log"
+	alloraMath "github.com/allora-network/allora-chain/math"
+	"github.com/allora-network/allora-chain/utils/fn"
+	"github.com/allora-network/allora-chain/x/emissions/keeper"
+	actorutils "github.com/allora-network/allora-chain/x/emissions/keeper/actor_utils"
+	inferencesynthesis "github.com/allora-network/allora-chain/x/emissions/keeper/inference_synthesis"
+	"github.com/allora-network/allora-chain/x/emissions/keeper/msgserver"
+	"github.com/allora-network/allora-chain/x/emissions/keeper/queryserver"
+	"github.com/allora-network/allora-chain/x/emissions/module"
+	"github.com/allora-network/allora-chain/x/emissions/types"
+	mintkeeper "github.com/allora-network/allora-chain/x/mint/keeper"
+	minttypes "github.com/allora-network/allora-chain/x/mint/types"
+)
+
+type TestSuite struct {
```

Comment:
This is exclusively used in the `x/emissions` module so maybe it should rely in it instead of this general package

Reply:
My idea was to also use it in x/mint, but it was agreed to skip refactoring mint for this PR

Reply:
I think the `x/mint` tests should rely on a mocked `x/emissions` keeper and not its actual implementation

Reply:
I don't see how we'll manage to inject mint's specific mocks in the test suite. I'm just afraid making a generic test suite for everything may be too much and counterproductive.

btw the current `x/mint` [test suite](https://github.com/allora-network/allora-chain/blob/33e1aeafaf7765c10bf5c89fff9f9dfe2fc9d673/x/mint/keeper/keeper_test.go#L27) is very good, it instantiates the mint keeper using mocked keepers which should be the standard way imho

Reply:
I haven't looked at x/mint closely, so you might be right. I don't mind moving the test util to be inside x/emissions for now